src/Engine/Combat/CombatEngine.java

```diff
@@ -25,10 +34,74 @@ public static BattleSummary calculateBattleResults( Unit attacker, Unit defender
    */
   public static BattleSummary simulateBattleResults( Unit attacker, Unit defender, GameMap map, int attackerX, int attackerY )
   {
-    // Set up our combat scenario.
-    BattleInstance params = new BattleInstance(attacker, defender, map, attackerX, attackerY, true);
+    return calculateBattleResults(attacker, defender, map, attackerX, attackerY, true);
+  }
+
+  /**
+   * Calculate and return the results of a battle.
+   * This will not actually apply the damage taken; this is done later in BattleEvent.
+   * Note that we assume the defender is where he thinks he is, but the attacker may move before attacking, so we take that coordinate explicitly.
+   * @param isSim Determines whether to cap damage at the HP of the victim in question
+   * @return A BattleSummary object containing all relevant details from this combat instance.
+   */
+  public static BattleSummary calculateBattleResults(Unit attacker, Unit defender, GameMap map, int attackerX, int attackerY, boolean isSim)
+  {
+    int defenderX = defender.x; // This variable is technically not necessary, but provides consistent names with attackerX/Y.
+    int defenderY = defender.y;
+    boolean attackerMoved = attacker.x != attackerX || attacker.y != attackerY;
+    int battleRange = Math.abs(attackerX - defenderX) + Math.abs(attackerY - defenderY);
+    WeaponModel attackerWeapon = attacker.chooseWeapon(defender.model, battleRange, attackerMoved);
+    WeaponModel defenderWeapon = null;
+    // Only attacks at point-blank range can be countered
+    if( 1 == battleRange )
+      defenderWeapon = defender.chooseWeapon(attacker.model, battleRange, false);
+
+    CombatContext context = new CombatContext(map, attacker, attackerWeapon, defender, defenderWeapon, battleRange, attackerX, attackerY);
+    
+    // If the attacker and defender get swapped, we want to still give a coherent BattleSummary.
```

Comment:
Does this happen often?

Reply:
Venge SCOP

Reply:
Ah, right. I'd be a little more descriptive in the comment then to clarify that they may be swapped *in the context object*, but our original variables will be fine.

To someone who know the system well this is obvious, but I never write comments for people who know the system well.